src/LLMProviders/embeddingManager.ts

```diff
@@ -9,43 +10,45 @@ import { OpenAIEmbeddings } from "langchain/embeddings/openai";
 export default class EmbeddingManager {
   private static instance: EmbeddingManager;
   private constructor(
-    private langChainParams: LangChainParams
+    private langChainParams: LangChainParams,
+    private encryptionService: EncryptionService,
   ) {}
 
   static getInstance(
-    langChainParams: LangChainParams
+    langChainParams: LangChainParams,
+    encryptionService: EncryptionService,
   ): EmbeddingManager {
     if (!EmbeddingManager.instance) {
-      EmbeddingManager.instance = new EmbeddingManager(langChainParams);
+      EmbeddingManager.instance = new EmbeddingManager(langChainParams, encryptionService);
     }
     return EmbeddingManager.instance;
   }
 
   getEmbeddingsAPI(): Embeddings | undefined {
+    const decrypt = (key: string) => this.encryptionService.getDecryptedKey(key);
```

Comment:
`const decrypt = this.encryptionService.getDecryptedKey` ?

why do we have to create an extra lamba for that